File: sample/src/main/java/com/moczul/sample/modifier/BasicAuthorizationHeaderModifier.java
package com.moczul.sample.modifier;

import com.moczul.ok2curl.Header;
import com.moczul.ok2curl.modifier.HeaderModifier;

public class BasicAuthorizationHeaderModifier implements HeaderModifier {

    private final Base64Decoder base64Decoder;

    public BasicAuthorizationHeaderModifier(Base64Decoder base64Decoder) {
        this.base64Decoder = base64Decoder;
    }

    @Override
    public boolean matches(Header header) {
        return "Authorization".equals(header.getName())
                && header.getValue().startsWith("Basic");
    }

    @Override
    public Header modify(Header header) {
        final String valueToDecode = header.getValue().replace("Basic", "").trim();
        final String decodedHeaderValue = base64Decoder.decode(valueToDecode);
        return new Header(header.getName(), decodedHeaderValue);
    }
}

File: sample/src/main/java/com/moczul/sample/modifier/BasicAuthorizationHeaderModifier.kt
package com.moczul.sample.modifier

import com.moczul.ok2curl.Header
import com.moczul.ok2curl.modifier.HeaderModifier

class BasicAuthorizationHeaderModifier(private val base64Decoder: Base64Decoder) : HeaderModifier {

    override fun matches(header: Header): Boolean {
        return "Authorization" == header.name && header.value.startsWith("Basic")
    }

    override fun modify(header: Header): Header {
        val valueToDecode = header.value.replace("Basic", "").trim()
        val decodedHeaderValue = base64Decoder.decode(valueToDecode)
        return Header(name = header.name, value = decodedHeaderValue)
    }
}